Tailor the return types of the various client.DB operations.

petermattis opened the pull request:
Single key operations now return a KeyValue instead of a
Result. Multi-key operations return a []KeyValue instead of a Result
because Result.Err is already checked for and returned in the error
return value.

Note: I decided to try this after having getting some more exposure to
client.DB. It removes the irritation of frequently having to access
Rows[0] at the expense of removing the uniformity of the return value
from the client.DB operations.

petermattis committed:
Tailor the return types of the various client.DB operations.

Single key operations now return a KeyValue instead of a
Result. Multi-key operations return a []KeyValue instead of a Result
because Result.Err is already checked for and returned in the error
return value.

tamird commented:
Looks like an improvement to me.

spencerkimball commented on examples/bank/bank.go:
```diff
@@ -114,7 +114,7 @@ func (bank *Bank) continuouslyTransferMoney(cash int64) {
 		transferMoney := func(runner client.Runner) error {
 			batchRead := &client.Batch{}
-			batchRead.Get(from, to)
+			batchRead.Get(from).Get(to)
```
Now where are you reading the "from" value?

petermattis replied:
Line 123: `batchRead.Results[0].Rows[0]`.

spencerkimball replied:
Ah, nevermind. I see you chained the gets. Missed that.

spencerkimball commented:
LGTM

petermattis commented:
@bdarnell? You had opinions about this before.

bdarnell commented on client/db.go:
```diff
@@ -252,78 +250,68 @@ func (db *DB) GetProto(key interface{}, msg gogoproto.Message) error {
 // encoding.BinaryMarshaler. value can be any key type or a proto.Message.
-func (db *DB) Put(key, value interface{}) (Result, error) {
-	return runOne(db, db.B.Put(key, value))
+func (db *DB) Put(key, value interface{}) (KeyValue, error) {
```
Now that we're tailoring return types, why does Put return a KeyValue at all? I think it should just return an error.

petermattis replied:
Yeah, good point. We never actually used the return value of `Put/CPut/Del`. Removed.

bdarnell commented on client/db.go:
```diff
@@ -252,78 +250,68 @@ func (db *DB) GetProto(key interface{}, msg gogoproto.Message) error {
 // encoding.BinaryMarshaler.
-func (db *DB) Del(keys ...interface{}) (Result, error) {
-	return runOne(db, db.B.Del(keys...))
+func (db *DB) Del(keys ...interface{}) ([]KeyValue, error) {
```
Again, why return a result here if it doesn't contain anything that wasn't in the input keys slice?

petermattis replied:
Removed.

bdarnell commented:
LGTM

petermattis committed:
Remove the KeyValue return value from Put/CPut/Del.

These return values were never once used and their info could be derived
from the arguments and error.
